src/conv/invokers/mlir_impl_gemm.cpp

```diff
@@ -69,6 +69,32 @@ auto PermuteDimStride(const std::vector<size_t>& dims, const std::vector<size_t>
     return std::make_tuple(sorted_dims, sorted_strides);
 };
 
+auto InsertGToDimStride(const std::string& layout,
```

Comment:
[Recommandation][Optimization] All function invocations look like:
```C++
std::make_tuple(D, S) =InsertGToDimStride(...D, S, ...);
```
I highly recommend changing the signature to:
```C++
void InsertGToDimStride(const std::string& layout,
                        const char dim,
                        const int group_count,
                        std::vector<size_t>& dims,
                        std::vector<size_t>& strides);
```
and making the function to return output values directly in `dims` and `strides`.

Reply:
Sure, I do see this redundancy but felt a bit more confusing for a different developer to see this argument is both input and output. Will update.